keras/callbacks.py

```diff
@@ -428,6 +429,48 @@ def on_epoch_end(self, epoch, logs=None):
                     self.model.save(filepath, overwrite=True)
 
 
+class DeadReluDetector(Callback):
+    def __init__(self, original_model, x_train):
+        if not isinstance(original_model, Sequential):
```

Comment:
In that case I'd check for Container or Model instead of Sequential.
the layers property is defined in Container (https://github.com/fchollet/keras/blob/master/keras/engine/topology.py#L1525), which is a base class of Model, which in turn is a base class of Sequential.
That way it will also work for models created with the functional API (that do not use Sequential)